AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m

```diff
@@ -236,4 +242,16 @@ - (void)checkIfTokenNeedsToBeRefreshed:(MSAuthTokenValidityInfo *)tokenValidityI
   }
 }
 
+- (void)finishInitialize {
+  if (self.resetAuthTokenRequired) {
+    [self setAuthToken:nil withAccountId:nil expiresOn:nil];
+  } else {
+    self.resetAuthTokenRequired = YES;
```

Comment:
@clpolet This is NOT correct. I'm going to revert your change.

Reply:
The problem here is if we decide to start some other module separately, it will keep resetting the token which doesn't seem necessary

Reply:
I reverted the changes and added test for this case. Lets wait for CI to tell us we're fine :)

Reply:
OK I was confused by this comment: https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1450#discussion_r271909023 . But yeah makes sense.